internal/rego/rego.go

```diff
@@ -108,12 +140,181 @@ func (r Rego) Path() string {
 
 // Parameters returns the list of parsed parameters
 func (r Rego) Parameters() []Parameter {
+	if len(r.annoParameters) > 0 {
+		return r.paramsAnnotationToLegacyFormat()
+	}
+
 	return r.parameters
 }
 
-// Severity returns the severity of the rego file.
-// When a rego file has multiple rules that are considered
-// to be different severities, the first rule is chosen.
+func (r Rego) paramsAnnotationToLegacyFormat() []Parameter {
+	var results []Parameter
+	for name, config := range r.annoParameters {
+		if config.Type == "array" {
+			results = append(results, Parameter{
+				Name:        name,
+				IsArray:     true,
+				Type:        config.Items.Schema.Type,
+				Description: config.Description,
+			})
+		} else {
+			results = append(results, Parameter{
+				Name:        name,
+				Type:        config.Type,
+				Description: config.Description,
+			})
+		}
+	}
+	return results
+}
+
+func (r Rego) AnnotationKindMatchers() []AnnoKindMatcher {
+	return r.annoKindMatchers
+}
+
+func (r Rego) AnnotationNamespaceMatchers() []string {
+	return r.annoNamespaceMatchers
+}
+
+func (r Rego) AnnotationExcludedNamespaceMatchers() []string {
+	return r.annoExcludedNamespaceMatchers
+}
+
+func (r Rego) AnnotationLabelSelectorMatcher() *metav1.LabelSelector {
+	return r.annoLabelSelector
+}
+
+func (r Rego) AnnotationParameters() map[string]apiextensionsv1.JSONSchemaProps {
+	return r.annoParameters
+}
+
+func (r Rego) GetAnnotation(name string) (any, bool) {
```

Comment:
this function seems to be unused, is this intended or am I missing something?

Reply:
It is used in the 'create' command to fetch the entire `matchers` key so that they can all be set at once without having to write code to set the Kinds/Labels/etc. matchers individually.

Reply:
oh, sorry, i’ve missed that